app/components/blacklight/gallery/slideshow_preview_component.rb

```diff
@@ -7,7 +7,8 @@ class SlideshowPreviewComponent < Blacklight::DocumentComponent
 
       def initialize(document:, document_counter: nil, **args)
         super(document: document, document_counter: document_counter, **args)
-        @document_counter = document_counter || @counter
+        # Items in JS start at 0. https://github.com/projectblacklight/blacklight-gallery/blob/main/app/assets/javascripts/blacklight_gallery/slideshow.js#L61
```

Comment:
Doesn't this depend on which version of ViewComponent you are running?
https://viewcomponent.org/CHANGELOG.html#v300

> BREAKING: Change counter variable to start iterating from 0 instead of 1.

Reply:
It does. Finally figured it out. Change has been pushed.